mission-control-client/src/App.js

```diff
@@ -1,20 +1,21 @@
-import React from "react";
+import React, { useEffect } from "react";
 import { Switch, Route } from "react-router-dom";
 import PrivateRoute from "./components/PrivateRoute";
 import Layout from "./components/layout/Layout";
 import "./styles/index.scss";
-import ReactGA from 'react-ga';
 import Registration from "./components/auth/Registration.js";
-import Login from "./components/auth/Login.js";
 import Home from "./components/test/Home";
 import Login from "./components/auth/Login";
 import DashboardHome from "./components/dashboard/admin-dashboard/DashboardHome";
+import embedAnalytics from './utils/embedAnalytics';
 
-//embedded Google Analytics for web metrics in React
-ReactGA.initialize('UA-111378465-2');
-ReactGA.pageview(window.location.pathname + window.location.search);
 
 function App() {
+  
+  useEffect(() => {
+    embedAnalytics();
+  }, []);
```

Comment:
is a dependency needed for this function to be called more then once?

Reply:
Yeah, you're right. I thought react-ga would watch the location but it does look like I'll need to pull in the useLocation hook and add it to the dependency array.